File: sources/paopao/Input/Control.cs
using System;

public struct Control
{
	public float getRealAccel()
	{
		return (!this.accelBrakeSwap) ? this.accel : this.brake;
	}

	public float getRealBrake()
	{
		return (!this.accelBrakeSwap) ? this.brake : this.accel;
	}

	public float getRealSteer()
	{
		return ((!this.wheelFlip && !this.wheelDevil) ? 1f : (-1f)) * this.steer;
	}

	public void Initialize()
	{
		this.accel = 0f;
		this.brake = 0f;
		this.accelBrakeSwap = false;
		this.steer = 0f;
		this.wheelFlip = false;
		this.wheelDevil = false;
		this.stayTime = 0f;
		this.steerAngle = 0f;
		this.oldSteerAngle = 0f;
	}

	public float accel;

	public float brake;

	public bool accelBrakeSwap;

	public float steer;

	public bool wheelFlip;

	public bool wheelDevil;

	public float stayTime;

	public float steerAngle;

	public float oldSteerAngle;
}

File: sources/roblox-lua-items-network/shared/kartEngine/KartMove/Control.lua
-- Control struct的Lua等效实现
local Control = {}
Control.__index = Control

function Control.new()
    local self = setmetatable({}, Control)
    self:Initialize()
    return self
end

function Control:getRealAccel()
    if not self.accelBrakeSwap then
        return self.accel
    else
        return self.brake
    end
end

function Control:getRealBrake()
    if not self.accelBrakeSwap then
        return self.brake
    else
        return self.accel
    end
end

function Control:getRealSteer()
    local multiplier = 1.0
    if self.wheelFlip or self.wheelDevil then
        multiplier = -1.0
    end
    return multiplier * self.steer
end

function Control:Initialize()
    self.accel = 0.0
    self.brake = 0.0
    self.accelBrakeSwap = false
    self.steer = 0.0
    self.wheelFlip = false
    self.wheelDevil = false
    self.stayTime = 0.0
    self.steerAngle = 0.0
    self.oldSteerAngle = 0.0
end

return Control